cep-XXXX.md

```diff
@@ -0,0 +1,201 @@
+
+Builds strings MUST only consist of ASCII letters, numbers, periods, plus symbols, and underscores.
+They MUST match this regex `^[a-zA-Z0-9_\.+]+$`.
+
+The maximum length of a build string MUST NOT exceed 64 characters.
+
+#### Artifact extensions
+
+Artifact extensions MUST only consist of lowercase ASCII letters, numbers and periods. They must start
+and end with a letter or a number. They MUST NOT include two consecutive periods. They MUST match this
+regex `^[a-z0-9](\.?[a-z0-9])*$`.
+
+The maximum length of a file extension MUST NOT exceed 16 characters.
+
+> The conda ecosystem currently recognizes two artifact extensions: `tar.bz2` and `conda`, versioned
+`v1` and `v2` respectively.
+
+#### Filenames
+
+Distributable conda artifacts MUST have a filename following this scheme:
+
+```text
+<package name>-<version string>-<build string>.<extension>
+```
+
+The maximum length of a filename MUST NOT exceed 192 characters.
+
+Virtual conda packages do not exist on disk and SHOULD NOT need filename standardization.
+
+#### Distribution strings
+
+A "distribution string" MAY be used to identify a package artifact without specifying the extension
+or the channel. It MUST match the following syntax:
+
+```text
+<package name>-<version string>-<build string>
+```
+
+Distribution strings apply to both distributable and virtual packages. They are used as the name of the
+directories where artifacts are extracted in the package cache, for example.
+
+> Note: Despite the similarity, distribution strings are not `MatchSpec`-like specifiers and
+> MUST NOT be used as such.
+
+### Identifying channels
+
+A conda channel is defined as a URL where one can find one or more `repodata.json` files arranged in one
+subdirectory (_subdir_) each. `noarch/repodata.json` MUST be present to consider the parent location a channel.
+
+#### Channel base URLs and names
+
+The base URL for the arbitrary location of a repodata file is defined as:
+
+```text
+<scheme>://[<authority>][/<path>/][/label/<label name>]/<subdir>/repodata.json
+```
+
+with `<scheme>`, `<authority>` and `<path>` defined by [RFC
+3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3.2).
+
+Taken the channel definition above, the base URL without trailing slashes is thus:
+
+```text
+<scheme>://[<authority>][/<path>/][/label/<label name>]
+```
+
+For example, given `https://conda.anaconda.org/conda-forge/noarch/repodata.json`, the part leading
+to `noarch/repodata.json` and thus base URL is `https://conda.anaconda.org/conda-forge/conda-forge`. For
+local repodata such as `file:///home/username/channel/noarch/repodata.json`, the
+channel base URL is `file:///home/username/channel`.
+
+For convenience, the channel _name_ is defined as the concatenation of `scheme`, `authority` and
+`path` components. The `<scheme>://` part MAY be omitted if `scheme` is one of `http`, `https` or
+`file`. At least one of `authority` or `path` SHOULD be present. In their absence, the channel name
```

Comment:
> The `<scheme>://` part MAY be omitted if `scheme` is one of `http`, `https` or
`file`.

Should we restrict this to just `file`? It seems odd to me that if I do something like `mkdir -pv conda.anaconda.org/mychannel/noarch`, multiple conda-compatible clients now have to figure out and agree if the schema-less argument `--channel conda.anaconda.org/mychannel/` refers to a local relative (`file://`-like) path or a HTTP(S) service.

Reply:
Added a new section for this only. Let me know what you think.

Reply:
Looks good. @chenghlee @jaimergp Do we know what rules `conda` uses when doing this type of inference? Do they match what is in the CEP? The rules don't have to match, but it'd be good to compare so we know what is currently done.

Reply:
Not really, I think.

For environment names (not channels), there are two CLI parts:

- `conda create` disambiguates via `-n` (name) and `-p` (path, prefix).
- `conda activate` does not, and instead relies on [this logic](https://github.com/conda/conda/blob/866f84041cb40a4122fe5e8154d67eb28817884a/conda/activate.py#L344-L353). This is similar to what we propose here.

For channels, it's a bit more [involved](https://github.com/conda/conda/blob/866f84041cb40a4122fe5e8154d67eb28817884a/conda/models/channel.py#L442-L472). They strip path components until there's a match in `custom_channels`, or give up and append it to `channel_alias` (conda.anaconda.org).

Reply:
ok LGTM.